rdbms/src/main/java/org/spine3/server/storage/jdbc/JdbcAggregateStorage.java

```diff
@@ -0,0 +1,228 @@
+import com.google.protobuf.Timestamp;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spine3.server.EntityId;
+import org.spine3.server.aggregate.Aggregate;
+import org.spine3.server.storage.AggregateStorage;
+import org.spine3.server.storage.AggregateStorageRecord;
+import org.spine3.server.storage.jdbc.util.*;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Iterator;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Lists.newLinkedList;
+import static java.lang.String.format;
+import static org.spine3.io.IoUtil.closeAll;
+import static org.spine3.server.Identifiers.idToString;
+import static org.spine3.server.storage.jdbc.util.Serializer.serialize;
+
+/**
+ * The implementation of the aggregate storage based on the RDBMS.
+ *
+ * @param <ID> the type of aggregate IDs. See {@link EntityId} for details.
+ * @see JdbcStorageFactory
+ * @author Alexander Litus
+ */
+/*package*/ class JdbcAggregateStorage<ID> extends AggregateStorage<ID> {
+
+    @SuppressWarnings({"UtilityClass", "DuplicateStringLiteralInspection", "ClassNamingConvention"})
+    private static class SQL {
+
+        /**
+         * Aggregate ID column name.
+         */
+        private static final String ID = "id";
+
+        /**
+         * Aggregate record column name.
+         */
+        private static final String AGGREGATE = "aggregate";
+
+        /**
+         * Aggregate event seconds column name.
+         */
+        private static final String SECONDS = "seconds";
+
+        /**
+         * Aggregate event nanoseconds column name.
+         */
+        private static final String NANOSECONDS = "nanoseconds";
+
+        private static final String INSERT =
+                "INSERT INTO %s " +
+                " (" + ID + ", " + AGGREGATE + ", " + SECONDS + ", " + NANOSECONDS + ") " +
+                " VALUES (?, ?, ?, ?);";
+
+        private static final String SELECT_BY_ID_SORTED_BY_TIME_DESC =
+                "SELECT " + AGGREGATE + " FROM %s " +
+                " WHERE " + ID + " = ? " +
+                " ORDER BY " + SECONDS + " DESC, " + NANOSECONDS + " DESC;";
+
+        private static final String CREATE_TABLE_IF_DOES_NOT_EXIST =
+                "CREATE TABLE IF NOT EXISTS %s (" +
+                    ID + " %s, " +
+                    AGGREGATE + " BLOB, " +
+                    SECONDS + " BIGINT, " +
+                    NANOSECONDS + " INT " +
+                ");";
+    }
+
+    private static final Descriptor RECORD_DESCRIPTOR = AggregateStorageRecord.getDescriptor();
+
+    private final DataSourceWrapper dataSource;
+
+    /**
+     * Iterators which are not closed yet.
+     */
+    private final Collection<DbIterator> iterators = newLinkedList();
+
+    private final IdHelper<ID> idHelper;
+
+    /**
+     * SQL queries.
+     */
+    private final String insertQuery;
```

Comment:
Why don't create a value object class, which accepts `tableName` and formats it into the query string?